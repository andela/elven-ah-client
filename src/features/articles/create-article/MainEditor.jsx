import React from 'react';
import { Editor } from '@tinymce/tinymce-react';
import PropTypes from 'prop-types';

const MainEditor = ({ handleChange, value, imageUploadHandler }) => (
  <Editor
    apiKey="3eytfdm1nnstcd6m2hzat9gpfvgg0ik5im0he4tpolgyqne9"
    cloudChannel="dev"
    value={value}
    onEditorChange={handleChange}
    name="body-editor"
    init={{
      menubar: false,
      inline: true,
      theme: 'inlite',
      mobile: { theme: 'mobile' },
      content_css: 'https://res.cloudinary.com/cj-odina/raw/upload/v1537743242/content_gvugw1.css',
      plugins: [
        'autolink',
        'codesample',
        'link',
        'linkchecker',
        'lists',
        'mediaembed',
        'textcolor',
        'image',
      ],
      selection_toolbar: 'bold italic | h2 h3 | blockquote quicklink | alignleft aligncenter alignright alignfull',
      file_picker_types: 'image',
      images_reuse_filename: true,
      images_upload_handler: imageUploadHandler,
    }}
  />
);

MainEditor.propTypes = {
  handleChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  imageUploadHandler: PropTypes.func.isRequired,
};

export default MainEditor;
