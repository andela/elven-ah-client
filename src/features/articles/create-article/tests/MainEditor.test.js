import React from 'react';
import { shallow } from 'enzyme';
import { Editor } from '@tinymce/tinymce-react';
import MainEditor from '../MainEditor';

const mockFunction = () => jest.fn();

describe('Main Editor component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<MainEditor
      value="<p>My new article</p>"
      handleChange={mockFunction}
      imageUploadHandler={mockFunction}
      handleBodyPlaceholderTextFocusIn={mockFunction}
      handleBodyPlaceholderTextFocusOut={mockFunction}
    />);
    expect(wrapper.find(Editor)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
