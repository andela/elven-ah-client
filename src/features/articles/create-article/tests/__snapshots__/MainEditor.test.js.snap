// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Editor component should render without errors 1`] = `
<Editor
  apiKey="3eytfdm1nnstcd6m2hzat9gpfvgg0ik5im0he4tpolgyqne9"
  cloudChannel="dev"
  init={
    Object {
      "content_css": "https://res.cloudinary.com/cj-odina/raw/upload/v1537743242/content_gvugw1.css",
      "file_picker_types": "image",
      "images_reuse_filename": true,
      "images_upload_handler": [Function],
      "inline": true,
      "menubar": false,
      "mobile": Object {
        "theme": "mobile",
      },
      "plugins": Array [
        "autolink",
        "codesample",
        "link",
        "linkchecker",
        "lists",
        "mediaembed",
        "textcolor",
        "image",
      ],
      "selection_toolbar": "bold italic | h2 h3 | blockquote quicklink | alignleft aligncenter alignright alignfull",
      "theme": "inlite",
    }
  }
  name="body-editor"
  onEditorChange={[Function]}
  onFocusIn={[Function]}
  onFocusOut={[Function]}
  value="<p>My new article</p>"
/>
`;
